src/Runner.Worker/JobExtension.cs

```diff
@@ -544,6 +557,32 @@ public void FinalizeJob(IExecutionContext jobContext, Pipelines.AgentJobRequestM
             }
         }
 
+        private async Task CheckDiskSpaceAsync(IExecutionContext context, CancellationToken token)
+        {
+            while (!token.IsCancellationRequested)
+            {
+                var workDirRoot = Directory.GetDirectoryRoot(HostContext.GetDirectory(WellKnownDirectory.Work));
+                var driveInfo = new DriveInfo(workDirRoot);
+                var freeSpaceInMB = driveInfo.AvailableFreeSpace / 1024 / 1024;
+                if (freeSpaceInMB < 100) // Add warning when disk is lower than 100 MB
```

Comment:
should we make this configurable? Maybe instead of a boolean free disk check we have it be an int and then if the int exists and is non-zero we check for that number of MB/KB?

Reply:
Is there any way we can add telemetry so we can discern what portion of abandons are due to this?

Reply:
`issue.Data[Constants.Runner.InternalTelemetryIssueDataKey] = Constants.Runner.LowDiskSpace;` will make this issue shows in Kusto